test/e2e/apps/job.go

```diff
@@ -1186,22 +1238,26 @@ func waitForJobEvent(ctx context.Context, config watchEventConfig) {
 	}
 }
 
-// waitForJobFailure uses c to wait for up to timeout for the Job named jobName in namespace ns to fail.
-func waitForJobFailure(ctx context.Context, c clientset.Interface, ns, jobName string, timeout time.Duration, reason string) error {
-	return wait.Poll(framework.Poll, timeout, func() (bool, error) {
+// waitForJobCondition uses c to wait for up to timeout for the Job named jobName in namespace ns to have the expected condition.
```

Comment:
It doesn't relate to this PR, but the line `uses c to wait` doesn't  make much sense (even though `c` is the func arg)

Reply:
Right, it seems like irrelevant technical detail. I dropped it and rephrased the comment a bit. PTAL.